plugins/Login/Controller.php

```diff
@@ -311,6 +311,15 @@ protected function authenticateAndRedirect($login, $password, $urlToRedirect = f
         // remove password reset entry if it exists
         $this->passwordResetter->removePasswordResetInfo($login);
 
+        $parsedUrl = parse_url($urlToRedirect);
+
+        // only use redirect url if host is trusted
+        if (!empty($parsedUrl['host']) && !Url::isValidHost($parsedUrl['host'])) {
+            $e = new \Piwik\Exception\Exception('The redirect URL host is not valid, it is not a trusted host. If this URL is trusted, you can allow this in your config.ini.php file by adding the line <i>trusted_hosts[] = "'.$parsedUrl['host'].'"</i> under <i>[General]</i>');
```

Comment:
This is probably unnecessary, but just in case, can we sanitize `$parsedUrl['host']`?

Reply:
changed that 👍